code/src/java/pcgen/gui3/PCGenMenuBar.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * PCGenMenuBar.java
+ * Copyright 2016 Connor Petty <cpmeister@users.sourceforge.net>
+ * 
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ * 
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ * 
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+ * 
+ * Created on Aug 26, 2016, 1:29:09 PM
+ */
+package pcgen.gui3;
+
+import java.util.List;
+import java.util.function.Function;
+import javafx.beans.binding.Bindings;
+import javafx.collections.ObservableList;
+import javafx.event.ActionEvent;
+import javafx.event.EventHandler;
+import javafx.scene.control.Menu;
+import javafx.scene.control.MenuBar;
+import javafx.scene.control.MenuItem;
+import javafx.scene.control.SeparatorMenuItem;
+import javafx.scene.input.KeyCombination;
+import pcgen.gui3.util.MappedList;
+
+/**
+ *
+ * @author Connor Petty <cpmeister@users.sourceforge.net>
+ */
+public class PCGenMenuBar extends MenuBar
+{
+
+	private final PCGenActions actions;
+
+	public PCGenMenuBar(PCGenActions actions)
+	{
+
+		this.actions = actions;
+
+		Menu fileMenu = new FileMenu();
```

Comment:
Can we move these into some kind of 'self registered classes' instead of hard coding it here? In fact we may want a general solution to "self registered classes".